### ISSUE TITLE
spider classify diday fails in CreateOutputStep

### ISSUE THREAD

**Issue by azazellochg** [I2PC/scipion#337]
I was following mix-and-match tutorial and in the end of run of Spider - classify Diday protocol I get a loooong error message:
``` bash
00058:   STARTED: createOutputStep, step 3
00059:     2016-02-22 18:08:14.402504
00060:   Traceback (most recent call last):
00061:     File "/home/sharov/soft/scipion/pyworkflow/protocol/protocol.py", line 167, in run
00062:       self._run()
00063:     File "/home/sharov/soft/scipion/pyworkflow/protocol/protocol.py", line 211, in _run
00064:       resultFiles = self._runFunc()
00065:     File "/home/sharov/soft/scipion/pyworkflow/protocol/protocol.py", line 207, in _runFunc
00066:       return self._func(*self._args)
00067:     File "/home/sharov/soft/scipion/pyworkflow/em/packages/spider/protocol/protocol_classify_base.py", line 138, in createOutputStep
00068:       self.buildDendrogram(True)
00069:     File "/home/sharov/soft/scipion/pyworkflow/em/packages/spider/protocol/protocol_classify_base.py", line 210, in buildDendrogram
00070:       return self._buildDendrogram(0, len(values)-1, 1, writeAverages)
00071:     File "/home/sharov/soft/scipion/pyworkflow/em/packages/spider/protocol/protocol_classify_base.py", line 272, in _buildDendrogram
00072:       self.addChildNode(node, leftIndex, m, 2*index, writeAverages, level)
00073:     File "/home/sharov/soft/scipion/pyworkflow/em/packages/spider/protocol/protocol_classify_base.py", line 216, in addChildNode
00074:       child = self._buildDendrogram(leftIndex, rightIndex, index, writeAverages, level+1)
00075:     File "/home/sharov/soft/scipion/pyworkflow/em/packages/spider/protocol/protocol_classify_base.py", line 273, in _buildDendrogram
00076:       self.addChildNode(node, m+1, rightIndex, 2*index+1, writeAverages, level)
00077:     File "/home/sharov/soft/scipion/pyworkflow/em/packages/spider/protocol/protocol_classify_base.py", line 216, in addChildNode
00078:       child = self._buildDendrogram(leftIndex, rightIndex, index, writeAverages, level+1)
...
...
02025:     File "/home/sharov/soft/scipion/pyworkflow/em/packages/spider/protocol/protocol_classify_base.py", line 216, in addChildNode
02026:       child = self._buildDendrogram(leftIndex, rightIndex, index, writeAverages, level+1)
02027:     File "/home/sharov/soft/scipion/pyworkflow/em/packages/spider/protocol/protocol_classify_base.py", line 248, in _buildDendrogram
02028:       node = DendroNode(index, height)
02029:     File "/home/sharov/soft/scipion/pyworkflow/em/packages/spider/protocol/protocol_classify_base.py", line 296, in __init__
02030:       graph.Node.__init__(self, 'class_%03d' % index)
02031:   RuntimeError: maximum recursion depth exceeded while calling a Python object
02032:   Protocol failed: maximum recursion depth exceeded while calling a Python object
02033:   FAILED: createOutputStep, step 3
```

**Comment by delarosatrevin** [comment]
I have got the same error following the Mix-and-Match.

**Comment by delarosatrevin** [comment]
This problem seems to arise when a lot of nodes in the classification docfile has ZERO height. 
It have been fixed in 67446cd19bba2a76e762a4dba1ad255b427f6142.
Still there is some problem with the total number of particles when showing Diday dendrogram.
